### ISSUE TITLE
InfluxDB 2.0.5: Changes to /debug endpoints

### ISSUE THREAD

**Issue by danxmoran** [influxdata/docs-v2#2239]
**PR:** influxdata/influxdb#20860

* A new `--pprof-disabled` flag to `influxd` allows users to disable all `/debug/pprof` endpoints
* `influxd upgrade` has been extended to convert the 1.x `http.pprof-enabled` setting into an equivalent 2.x `pprof-disabled` setting in config
* When enabled, a new `/debug/pprof/all` endpoint allows for collecting all profiling information in a single request
* The `/debug/vars` endpoint has been removed. The `/metrics` endpoint is/was meant to replace it in 2.x, but it suck into our previous builds because some dependency was enabling it.

**Comment by pierwill** [comment]
Thanks, @danxmoran. A few questions:

- Has `/debug/vars` been removed from 1.x or 2.0? Looks to me like the docs don't mention this endpoint for 2.0 (`cd docs-v2/content/influxdb/v2.0; rg "debug/vars"`), so we might not need any docs changes for this change.
- How would a user go about enabling `/debug/pprof/all`?

**Comment by danxmoran** [comment]
`/debug/vars` was removed from 2.0. I'm not surprised it wasn't documented, AFAIK it was included in previous 2.0 builds by mistake.

`/debug/pprof/all` is enabled by default along with the rest of the profiling endpoints, and disabled when `--pprof-disabled` is passed to `influxd`. Users can `curl` it, for example:
```
curl localhost:8086/debug/pprof/all
```
I believe the `/all` endpoint is something we added to help with bug reports against `influxdb` (it's an easy way for users to send us a pseudo-snapshot of their running system when it's in a bad state). For general info about the profiling tools, it might be useful to link to [this blog post](https://blog.golang.org/pprof).
